src/main/scala/org/portablescala/sbtplatformdeps/PlatformDepsBuilders.scala

```diff
@@ -0,0 +1,52 @@
+package org.portablescala.sbtplatformdeps
+
+import sbt._
+
+import scala.language.experimental.macros
+import scala.reflect.macros.Context
+
+final class PlatformDepsGroupID private[sbtplatformdeps] (
+    private val groupID: String) {
+  def %%%(artifactID: String): PlatformDepsGroupArtifactID =
+    macro PlatformDepsGroupID.cross_binary_impl
+}
+
+object PlatformDepsGroupID {
+  /** Internal, used by the macro implementing [[PlatformDepsGroupID.%%%]].
+   *
+   *  Use [[PlatformDepsPlugin.autoImport.platformDepsCrossVersion]] instead.
+   */
+  val platformDepsCrossVersion: SettingKey[CrossVersion] =
+    PlatformDepsPlugin.autoImport.platformDepsCrossVersion
+
+  /** Internal, used by the macro implementing [[PlatformDepsGroupID.%%%]].
+   *
+   *  Use:
+   *  {{{
+   *  ("a" % artifactID % revision).cross(cross)
+   *  }}}
+   *  instead.
+   */
+  def withCross(groupID: PlatformDepsGroupID, artifactID: String,
+      cross: CrossVersion): PlatformDepsGroupArtifactID = {
+    require(artifactID.nonEmpty, "Artifact ID must not be empty")
+    new PlatformDepsGroupArtifactID(groupID.groupID, artifactID, cross)
+  }
+
+  def cross_binary_impl(c: Context { type PrefixType = PlatformDepsGroupID })(
```

Comment:
If we really just want to build on 2.12, use `BlackboxContext` here? Also, in this case, turn on fatal warnings.

Reply:
We actually do build on 0.13 with 2.10. Do we need that, I'm afraid.